Allow specifying a colon-separated list of directories as a search
path for mechanism plugins.

--- lib/common.c.orig	2012-10-12 10:05:48.000000000 -0400
+++ lib/common.c	2012-12-04 13:54:10.000000000 -0500
@@ -1567,6 +1567,10 @@
   }
 }
 
+#ifndef PLUGINPATH
+#define PLUGINPATH	PLUGINDIR
+#endif
+
 static int
 _sasl_getpath(void *context __attribute__((unused)),
               const char ** path_dest)
@@ -1587,12 +1591,12 @@
         /* NB: On Windows platforms this value is always allocated */
         default_plugin_path = _sasl_get_default_win_path(context,
                                                          SASL_PLUGIN_PATH_ATTR,
-                                                         PLUGINDIR);
+                                                         PLUGINPATH);
 #else
         /* NB: On Unix platforms this value is never allocated */
         path = _sasl_get_default_unix_path(context,
                                            SASL_PATH_ENV_VAR,
-                                           PLUGINDIR);
+                                           PLUGINPATH);
 
         res = _sasl_strdup(path, &default_plugin_path, NULL);
 #endif
